url8d3c4d17f5_catseye_JaC64_tgz-pJ8-JaC64J8/src/JaC64.java

```diff
@@ -0,0 +1,300 @@
+    reader.setCPU(cpu);
+    cpu.getDrive().setReader(reader);
+
+    C64Win = new JFrame("JaC64 - A Java C64 Emulator");
+    C64Win.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE); 
+    JMenuBar jbar = new JMenuBar();
+    C64Win.setJMenuBar(jbar);
+    JMenu filem;
+    JMenuItem mi;
+    jbar.add(filem = new JMenu("File"));
+    filem.add(mi = new JMenuItem("Open File/Disk"));
+    mi.addActionListener(this);
+    filem.add(load = new JMenuItem("Load File"));
+    load.addActionListener(this);
+    filem.add(mi = new JMenuItem("Reset"));
+    mi.addActionListener(this);
+    filem.add(mi = new JMenuItem("Hard Reset"));
+    mi.addActionListener(this);
+    filem.add(mi = new JMenuItem("About JaC64"));
+    mi.addActionListener(this);
+
+    jbar.add(filem = new JMenu("Settings"));
+    JMenu subm;
+    filem.add(subm = new JMenu("Color Set"));
+    createRadioMenu(subm, new String[] {"Color Set 1 - JaC64 original",
+        "Color Set 2 - darker", "Color Set 3 - softer", 
+        "Color Set 4 - Win VICE"}, 0);
+
+    filem.add(subm = new JMenu("SID Emulation"));
+    createRadioMenu(subm, SID_TYPES, 0);
+    
+    filem.add(subm = new JMenu("Joystick Port"));
+    createRadioMenu(subm, JOYSTICK, 0);
+    
+    C64Win.setBackground(Color.black);
+    C64Win.setForeground(Color.black);
+    C64Win.setLayout(new BorderLayout());
+    C64Win.getContentPane().add(scr.getScreen(),
+				BorderLayout.CENTER);
+    C64Win.setFocusable(true);
+
+    C64Win.pack(); //	C64Scr.setSize(380,300);
+    C64Win.setSize(386 * 2 + 10, 284 * 2 + 70);
+    C64Win.setResizable(true);
+    C64Win.setVisible(true);
+    
+    KeyboardFocusManager.
+      getCurrentKeyboardFocusManager().addKeyEventDispatcher(this);
+
+    // Setup disk sounds
+    AudioClip trackSound = null;
+    AudioClip motorSound = null;
+    URL url = getClass().getResource("sounds/track.wav");
+    if (url != null) trackSound = Applet.newAudioClip(url);
+    url = getClass().getResource("sounds/motor.wav");
+    if (url != null) motorSound = Applet.newAudioClip(url);
+    scr.setSounds(trackSound, motorSound);
+    c64Canvas = (KeyListener) scr.getScreen();
+  }
+
+  private void createRadioMenu(JMenu subm, String[] names, int selected) {
+    ButtonGroup group = new ButtonGroup();
+    JRadioButtonMenuItem mi;
+    for (int i = 0; i < names.length; i++) {
+      subm.add(mi = new JRadioButtonMenuItem(names[i]));
+      if (i == selected)
+        mi.setSelected(true);
+      mi.addActionListener(this);
+      group.add(mi);      
+    }
+  }
+  
+  public boolean dispatchKeyEvent(KeyEvent e) {
+    if (C64Win.isFocused()) {
+      if (e.getID() == KeyEvent.KEY_PRESSED) {
+        c64Canvas.keyPressed(e);
+      } else if (e.getID() == KeyEvent.KEY_RELEASED) {
+        c64Canvas.keyReleased(e);
+      } else if (e.getID() == KeyEvent.KEY_TYPED) {
+        c64Canvas.keyTyped(e);
+      }
+      return true;
+    }
+    return false;
+  }
+
+  public void actionPerformed(ActionEvent ae) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 14. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.